x-pack/packages/kbn-elastic-assistant/impl/assistant/api.tsx

```diff
@@ -153,14 +186,24 @@ export const fetchConnectorExecuteAction = async ({
           }
         : undefined;
 
+    onNewReplacements(response.replacements ?? {});
+
     return {
-      response: assistantLangChain ? getFormattedMessageContent(response.data) : response.data,
+      response: hasParsableResponse({
+        alerts,
+        assistantLangChain,
+        ragOnAlerts,
+      })
+        ? getFormattedMessageContent(response.data)
+        : response.data,
       isError: false,
       isStream: false,
       traceData,
     };
   } catch (error) {
-    const reader = error?.response?.body?.getReader();
+    const getReader = error?.response?.body?.getReader;
+    const reader =
+      isStream && typeof getReader === 'function' ? getReader.call(error.response.body) : null;
```

Comment:
note: this is the fix for the existing bug on `main` where errors were not bubbling up. Ty for the fix!